src/Request.js

```diff
@@ -245,17 +246,23 @@ export default class Request {
 
     retry () {
 
-        if (this.retries++ < this._max_retry) {
+        if (this.last_error) {
+            this.errors.push(this.last_error);
+        }
+
+        if (this.retries + 1 < this._max_retry) {
```

Comment:
Why would that be a problem? Won't that have the same behavior as current implementation, since we don't return until L259?